src/codemagic/tools/firebase_app_distribution/firebase_action.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from abc import ABCMeta
+from abc import abstractmethod
+from typing import List
+
+from codemagic.firebase.client import FirebaseClient
+from codemagic.firebase.resource_managers.resource_manager import ResourceManager
+from codemagic.firebase.resources import Release
+
+
+class FirebaseAction(metaclass=ABCMeta):
```

Comment:
Some more confusion with file and class names. Wouldn't `FirebaseAppDistributionAction` be better?